ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -604,16 +628,24 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
     renderPreviewDataTable() {
       const filteredPreviewData =
           this.state.previewList.filter(
-            preview => fp.contains(preview.domain, this.state.selectedPreviewDomain))[0];
-      return <DataTable ref={el => this.dt = el} key={this.state.selectedPreviewDomain}
-                        scrollable={true} style={{width: '100%'}}
-                        value={this.getDataTableValue(filteredPreviewData.values)}>
-        {filteredPreviewData.values.map(value =>
-          <Column key={value.value} header={this.getHeaderValue(value)}
-                  headerStyle={{textAlign: 'left', width: '5rem'}} style={{width: '5rem'}}
-                  field={value.value}/>
-        )}
-      </DataTable>;
+            preview => preview.domain === this.state.selectedPreviewDomain)[0];
+      const domainDisplayed = this.state.selectedPreviewDomain.toString().toLowerCase();
+      return filteredPreviewData.values.length > 0 ?
+        <DataTable ref={el => this.dt = el} key={this.state.selectedPreviewDomain}
```

Comment:
i think this will be quicker to read if we pull out this rendering logic into a function that takes in a `filteredPreviewData` object and returns the JSX.

 I think it would make the overview of this render function easy to read at a glance since it simplifies down to 
- find your data
- if it exists, render it
  - if not, loading message